include/glow/ExecutionEngine/ExecutionEngine.h

```diff
@@ -159,6 +159,17 @@ void runBatch(ExecutionEngine &EE, PlaceholderBindings &bindings,
               llvm::ArrayRef<Placeholder *> ph, llvm::ArrayRef<Tensor *> inputs,
               llvm::StringRef name = "");
 
+/// Runs \p numMinibatchRuns iterations of the compiled function called \p name.
+/// The method updates a global counter and future invocations of this method
+/// continue running iterations of the batch at the next available slice.
+/// Each sample output is copmared VS ground-truth \p labelsInput, and
+/// \p numCorrectEstimates counter is updated correspondingly.
+void evalBatch(ExecutionEngine &EE, PlaceholderBindings &bindings,
+               size_t numMinibatchRuns, size_t &sampleCounter,
+               Placeholder *inputPH, Placeholder *outputPH,
+               Tensor &samplesInput, Tensor &labelsInput, llvm::StringRef name,
+               size_t &numCorrectEstimates,
+               std::vector<Tensor> *estimates = nullptr);
```

Comment:
How about two methods, one just returns back TWO tensors (output and labels) through callback, and second method given two tensors "visualizes" and calculating accuracy?

Reply:
Following an offline chat, modified evalBatch to take a callback (invoked on each sample) which allows the user to control results accumulation and visualization.